modules/nf-core/gmmdemux/main.nf

```diff
@@ -0,0 +1,68 @@
+
+process GMMDEMUX {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gmm-demux:0.2.2.3--pyh7cba7a3_0':
+        'biocontainers/gmm-demux:0.2.2.3--pyh7cba7a3_0' }"
+
+    input:
+    tuple val(meta), path(hto_matrix),val(hto_names)
+    val type_report
+    val summary_report
+    val skip
+
+    output:
+    tuple val(meta), path("${meta.id}/barcodes.tsv.gz"                          ), emit: barcodes
+    tuple val(meta), path("${meta.id}/*.mtx.gz"                                 ), emit: matrix
+    tuple val(meta), path("${meta.id}/features.tsv.gz"                          ), emit: features
+    tuple val(meta), path("${meta.id}/classification_report_${meta.id}"         ), emit: classification_report
+    tuple val(meta), path("summary_report_${meta.id}.txt"                       ), emit: summary_report, optional: true
```

Comment:
I changed this, but somehow I get an error: ```
Caused by:                                                                                                                                                            │
│     Missing output file(s) `null/barcodes.tsv.gz` expected by process `GMMDEMUX (test)`   
```
I imagine this is because prefix is defined as meta.id down below, before the script is executed: `def prefix         = task.ext.prefix ?: "${meta.id}"` is that possible?

Reply:
That in itself would be expected behaviour. We want the tool to use the meta.id by default and only overwrite it if a different prefix is specified. 
I had a look at the tests and I'm not sure why the $prefix would be null though as you specified the meta.id. 🤔 I can't see the issue right now, will have to have a more detailed look later. It looks like something is going wrong with the prefix for sure.

Reply:
Oh, try removing the `def` in front of the prefix definitions, that should make the variable global and therefore be able to be seen for the output channels.